test-programs/compressed-token-test/tests/test.rs

```diff
@@ -243,27 +248,424 @@ async fn test_mint_to<const MINTS: usize, const ITER: usize>() {
+            authority: payer_2.pubkey(),
+            cpi_authority_pda: get_cpi_authority_pda().0,
+            mint: mint_2.clone(),
+            token_pool_pda: mint_pool_1,
+            token_program: anchor_spl::token::ID,
+            light_system_program: light_system_program::ID,
+            registered_program_pda: light_system_program::utils::get_registered_program_pda(
+                &light_system_program::ID,
+            ),
+            noop_program: Pubkey::new_from_array(
+                account_compression::utils::constants::NOOP_PUBKEY,
+            ),
+            account_compression_authority: light_system_program::utils::get_cpi_authority_pda(
+                &light_system_program::ID,
+            ),
+            account_compression_program: account_compression::ID,
+            merkle_tree: merkle_tree_pubkey,
+            self_program: light_compressed_token::ID,
+            system_program: system_program::ID,
+        };
+        let instruction = Instruction {
+            program_id: light_compressed_token::ID,
+            accounts: accounts.to_account_metas(Some(true)),
+            data: instruction_data.data(),
+        };
+        rpc.create_and_send_transaction(&[instruction], &payer_2.pubkey(), &[&payer_2])
+            .await
+            .unwrap_err();
+    }
+    // 5. Invalid CPI authority.
+    {
+        let invalid_cpi_authority_pda = Keypair::new();
+        let accounts = light_compressed_token::accounts::MintToInstruction {
+            fee_payer: payer_2.pubkey(),
+            authority: payer_2.pubkey(),
+            cpi_authority_pda: invalid_cpi_authority_pda.pubkey(),
+            mint: mint_1.clone(),
+            token_pool_pda: mint_pool_1,
+            token_program: anchor_spl::token::ID,
+            light_system_program: light_system_program::ID,
+            registered_program_pda: light_system_program::utils::get_registered_program_pda(
+                &light_system_program::ID,
+            ),
+            noop_program: Pubkey::new_from_array(
+                account_compression::utils::constants::NOOP_PUBKEY,
+            ),
+            account_compression_authority: light_system_program::utils::get_cpi_authority_pda(
+                &light_system_program::ID,
+            ),
+            account_compression_program: account_compression::ID,
+            merkle_tree: merkle_tree_pubkey,
+            self_program: light_compressed_token::ID,
+            system_program: system_program::ID,
+        };
+        let instruction = Instruction {
+            program_id: light_compressed_token::ID,
+            accounts: accounts.to_account_metas(Some(true)),
+            data: instruction_data.data(),
+        };
+        let result = rpc
+            .create_and_send_transaction(&[instruction], &payer_2.pubkey(), &[&payer_2])
+            .await;
+        assert_rpc_error(
+            result,
+            0,
+            anchor_lang::error::ErrorCode::ConstraintSeeds.into(),
```

Comment:
can we add a specific error here as well?

Reply:
Same as above. There is no custom error support with `@` for seeds, and the seed predicate looks like:

```rust
    #[account(seeds = [CPI_AUTHORITY_PDA_SEED], bump)]
    pub cpi_authority_pda: UncheckedAccount<'info>,

```